File: baekjoon/1259.js
//let fs = require('fs');
//let input = fs.readFileSync('/dev/stdin').toString().split('\n');

const input = [
'121',
'1231',
'12421',
'12345654321',
'1234553421',
'3214894153156445614561231',
'12345678987654321',
'0'
]

for(i=0; i<input.length-1; i++){
  let q = input[i].split('');
  let p = q.length;
  let bo = true;
  let j = 0;

  while(j<p/2){
    if(q[j]!=q[p-1-j]){
      bo = false;
      break;
    }
    j++;
  }
  if(bo){
    console.log("yes");
  } else {
    console.log("no");
  }
}

// 맞는데 왜 틀리다는겨;;

File: baekjoon/1259.py
n = ['121', '1231', '12421', '0']

while True:
    n = input()
    if (n == '0'):
        break
    for i in range(len(n)):
        if (i == len(n) - 1):
            print('yes')
            break
        if (n[i] != n[len(n) - 1 - i]):
            print('no')
            break
